src/controllers/wish.controller.ts

```diff
@@ -1,66 +1,93 @@
-  const productId = req.params.product_id;
-  const user: User = req.user as User;
-  const userId: string = user.dataValues.id as string;
-
-  if (productId != null) {
-    const data = {
-      productId,
-      userId
-    };
-
-    const exWish = await wishServices.getSingleWish(data);
-    if (exWish == null) {
-      const wish = await wishServices.createWish(data);
-
-      if (wish != null) {
+  try {
+    const { productId } = req.body;
+    const isValid: boolean = isValidUUID(productId);
+    const user: User = req.user as User;
+    const userId: string = user.dataValues.id as string;
+    if (isValid && productId != null) {
+      const data = { productId, userId };
+      const exWish = await wishServices.getSingleWish(data);
+      if (exWish == null) {
+        await wishServices.createWish(data);
         res.status(200).send({
           message: "product added to wishlist",
-          data: { productId, userId }
+          data
         });
       } else {
-        res.status(500).send({ error: "Could not create a wish" });
+        await wishServices.deleteWish(data);
+        res.status(200).send({ message: "Product removed into wishlist" });
       }
-    } else {
-      await wishServices.deleteWish(data);
-      res.status(200).send({ message: "Product removed into wishlist" });
     }
-  } else {
-    res.status(500).send({ error: "Could not get a product" });
-  }
-};
-
-export const getProductWishes = async (req: Request, res: Response) => {
-  const productId = req.params.product_id;
-  if (productId != null) {
-    const wishes = await wishServices.getProductWishes(productId);
-
-    if (wishes != null) {
-      res.status(200).send({ message: "Product wishes", data: wishes });
-    } else {
-      res
-        .status(200)
-        .send({ message: "Product wishes is empty", data: wishes });
-    }
-  } else {
-    res.status(400).send({ message: "please provide valid product id" });
+  } catch (error) {
+    res.status(500).send({ error: "Could add product wishes" });
   }
 };
 
 export const getUserWishes = async (req: Request, res: Response) => {
   const user: User = req.user as User;
   const userId: string = user.dataValues.id as string;
+  const userRole: string = user.dataValues.roleId as string;
 
   if (userId != null) {
-    const wishes = await wishServices.getUserWishes(userId);
-    if (wishes != null) {
-      res.status(200).send({ message: "product wishlist", data: wishes });
-    } else {
-      res.status(200).send({ message: "product wish list is empty" });
+    const role = await Role.findByPk(userRole);
+    let wishesData: any[] = [];
+
+    if (role?.dataValues.name == "buyer") {
+      const wishes: any = await wishServices.getUserWishes(userId);
+      wishesData = await Promise.all(
+        wishes.map(async (wish: any) => {
+          const { productId, userId, updatedAt, ...wishData } = wish.dataValues;
+          const product = await Product.findByPk(productId, {
```

Comment:
Do not call database inside the loop. Instead, generate an array of product ids from wishes object and use that array in `Product.findAll` method, this way you will get information of the products matching the ids in the array without looping through database.

Reply:
Resolved